src/scripts/items/FluteItem.ts

```diff
@@ -0,0 +1,57 @@
+///<reference path="Item.ts"/>
+class FluteItem extends Item {
+
+    type: GameConstants.FluteItemType;
+
+    constructor(
+        type: GameConstants.FluteItemType,
+        description: string,
+        basePrice: number,
+        currency: undefined,
+        public gemTypes: (keyof typeof PokemonType)[],
+        displayName?: string,
+        public multiplierType?: keyof typeof MultiplierType,
+        public multiplyBy?: number
+    ) {
+        super(GameConstants.FluteItemType[type], basePrice, currency, undefined, displayName, description, 'fluteItem');
+        this.type = type;
+    }
+
+    use(): boolean {
+        fluteEffectRunner.toggleEffect(this.name);
+        return true;
+    }
+
+    checkCanUse(): boolean {
+        if (App.game.challenges.list.disableGems.active()) {
```

Comment:
The "Flutes are disabled" banner also displays when `disableBattleItems.active()`, you should include an if check for that one too.